### PR TITLE
Time format

### PR DESCRIPTION
**Pull request by bhavyaab**
## Description
- The time input is editable so the time can be changed
- The time is in 12 hours (am/pm) format
- Clicking the left arrow decreases the time by 30 min
- Clicking the right arrow increases the time by 30 min

## Test Plan
- run yarn webpack
- go to http://localhost:4000/verification/pregnancy-resource-center
- click 'verify next resource' button
- scroll down to hours component
- click on arrows left or right
   - if left arrow is clicked and time is missing then it will auto-fill 9:00 am
   - if right arrow is clicked and time is missing then it will auto-fill 5:00 pm
   - now clicking left arrow will decrease time by 30 min
   - and clicking right arrow will increase the time by 30 min
   - when it will reach 12:00 it will change to am to pm if it was am before otherwise, it will change to am

### CODE REVIEW

File: client/js/verification-portal/pregnancy-resource-center/get-resource-to-verify-button.js [hunk 1]
```diff
@@ -12,10 +12,11 @@ const convertNumberToTimeFormat = timeNumber => {
 	if (!timeNumber) {
 		return null
 	}
-
 	const timeString = String(timeNumber).padStart(4, '0')
-
-	return `${timeString.slice(0, 2)}:${timeString.slice(2, 4)}`
+	const hours = parseInt(`${timeString.slice(0, 2)}`)
+	let am = ''
+	am = hours > 12 ? 'pm' : 'am'
```
Comment (karenmrose) [inline client/js/verification-portal/pregnancy-resource-center/get-resource-to-verify-button.js]: @bhavyaab This can be a confusing variable name because `am` could actually equal `'pm'`. The actual name is `meridiem`. What do you think of using that as a variable name instead?

Also, these 2 lines can be shortened to:
`const meridiem = hours > 12 ? 'pm' : 'am'`

File: client/js/verification-portal/pregnancy-resource-center/form.js [hunk 2]
```diff
@@ -260,37 +261,37 @@ class VerificationPortalForm extends Component {
...
-					<Field name="hours[6].open" component={Input} type="time" />
-					<Field name="hours[6].close" component={Input} type="time" />
+					<Field name="hours[6].open" component={Time} placeholder={'Open'} />
+					<Field name="hours[6].close" component={Time} placeholder={'Close'} />
 				</div>
 
 				<div className={classes.parent}>
 					<Field
-						name="verifiedData.hours.verified"
+						name="verifiedTimeData.hours.verified"
```
Comment (karenmrose) [inline client/js/verification-portal/pregnancy-resource-center/form.js]: @bhavyaab What is the reasoning for this change? All of the values that need to be verified are stored in the `verifiedData` object.

File: client/js/verification-portal/pregnancy-resource-center/form.js [hunk 3]
```diff
@@ -260,37 +261,37 @@ class VerificationPortalForm extends Component {
 					<Heading text="HOURS" />
 					<Spacer height="50px" />
 					<label>Sunday Hours</label>
-					<Field name="hours[0].open" component={Input} type="time" />
-					<Field name="hours[0].close" component={Input} type="time" />
+					<Field name="hours[0].open" component={Time} placeholder={'Open'} />
```
Comment (karenmrose) [inline client/js/verification-portal/pregnancy-resource-center/form.js]: @bhavyaab We should be able to capture the placeholder in the `Time` component based on the field name so that we don't have to repeat it over and over in this file.

File: client/js/verification-portal/components/time.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+					onClick={() => timeSpin(input.value, input.name, 'down')}
+				>
+					{'>'}
+				</span>
+			</div>
+		)
+	}
+}
+const styles = {
+	warper: {
```
Comment (karenmrose) [inline client/js/verification-portal/components/time.js]: @bhavyaab Is the intended name here `wrapper`?

File: client/js/verification-portal/components/time.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+			} else {
+				if (!hr[i + 1]) {
+					i = -1
+					current[1] == 'am' ? (current[1] = 'pm') : (current[1] = 'am')
+				}
+				current[0] = hr[i + 1]
+			}
+			current = current.join(' ')
+			changeFieldValue(name, current)
+		}
```
Comment (karenmrose) [inline client/js/verification-portal/components/time.js]: @bhavyaab Nice job getting this to work! 👏

We're going to need to clean this function up though so that it is more readable and maintainable. 😄 I'm having a hard time figuring out what is actually going on here. A couple of suggestions:
- This function can be pulled out of the class (we should avoid nested functions as much as possible)
- The naming of variables is extremely important in knowing what the value represents and helps a lot with readability. I would encourage you to think of more descriptive (and probably longer) variable names.
- If possible, nested `if` statements should be avoided. By refactoring the code to remove the nested ifs, it's likely it will become more readable.

Let me know if you have any questions or would like any help in cleaning this up, I would be more than happy to help!

File: client/js/verification-portal/components/time.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import React from 'react'
+import injectSheet from 'react-jss'
+import { connect } from 'react-redux'
+import { change } from 'redux-form'
+
+class Time extends React.Component {
```
Comment (karenmrose) [inline client/js/verification-portal/components/time.js]: I don't think there is any reason this needs to be a class. What do you think of using this style?
https://github.com/HelpAssistHer/help-assist-her/blob/master/client/js/components/button.js#L5

File: client/js/verification-portal/pregnancy-resource-center/form.js (hunk 2 above)
Comment (bhavyaab) [reply client/js/verification-portal/pregnancy-resource-center/form.js]: It was by mistake.

File: client/js/verification-portal/pregnancy-resource-center/get-resource-to-verify-button.js [hunk 7]
```diff
@@ -12,10 +12,11 @@ const convertNumberToTimeFormat = timeNumber => {
...
+	return `${hours > 12 ? hours - 12 : hours}:${timeString.slice(2, 4)}  ${am}`
```
[the remark this answers is not in the document]
Comment (bhavyaab) [reply client/js/verification-portal/pregnancy-resource-center/get-resource-to-verify-button.js]: agreed, going to change it.

File: client/js/verification-portal/components/time.js (hunk 6 above)
Comment (bhavyaab) [reply client/js/verification-portal/components/time.js]: Is it possible to dispatch changeFieldValue without making it class?

File: client/js/verification-portal/components/time.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+	wrapper: {
+		color: '#000',
+		padding: '2.5px',
+		'border-bottom': '1px solid #979797',
+		width: '105px',
+	},
+	input: {
+		height: '17px',
+		width: '84px',
+		'font-family': 'Century Gothic',
```
Comment (karenmrose) [inline client/js/verification-portal/components/time.js]: So for whatever reason this font does not actually match the designs. It's not the most ideal font. I've been using `'font-family': 'sans-serif'` and that seems to better match the designs.

**Comment by karenmrose** [comment]
@bhavyaab Really nice job cleaning this up! 👏 

I left a couple of minor comments. Also, it looks like when you try to save the hours, it's saving as `null` in the database.

screen shot 2018-08-02 at 12 27 25 pm

File: client/js/verification-portal/components/time.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import injectSheet from 'react-jss'
+import { connect } from 'react-redux'
+import { change } from 'redux-form'
+
+let i = 0
```
[the remark this answers is not in the document]
Comment (bhavyaab) [reply client/js/verification-portal/components/time.js]: Yes, we can, but defining the variable in every function call is memory consuming, and not a good practice.

File: client/js/verification-portal/components/time.js (hunk 8 above)
Comment (bhavyaab) [reply client/js/verification-portal/components/time.js]: It is been copied from inVision CSS, there it is been used.

File: client/js/verification-portal/components/time.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (karenmrose) [reply client/js/verification-portal/components/time.js]: Hmm. Yeah technically it would consume more memory, but the amount is so small that I can't imagine it would make any difference in a noticeable way. In my opinion, it's much better to have more readable code than it is to address performance issues that don't cause actual noticeable differences (i.e the user actually notices the page is slower). Thoughts?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -3,15 +3,16 @@
 1.  Request access to this repo
 
 1.  Pull the latest master
-    - On the `master` branch: `git pull`
+
+    * On the `master` branch: `git pull`
```
Comment (karenmrose) [inline CONTRIBUTING.md]: @bhavyaab Can you revert the changes to this file from this PR?

Comment (bhavyaab) [reply CONTRIBUTING.md]: I can't because it is done by auto linting on commit. Linting rules need to be fixed.

Comment (karenmrose) [reply CONTRIBUTING.md]: @bhavyaab The linting doesn't actually change any code. It just reads it and checks to make sure things pass. Is your IDE set to make changes?

**Comment by karenmrose** [comment]
@bhavyaab The tests are still failing. The broken test has been fixed on master so if you have the latest, then it should pass on this branch too. Make sure you checkout master and do a `git pull`

File: CONTRIBUTING.md (hunk 10 above)
Comment (karenmrose) [reply CONTRIBUTING.md]: Oh man, I think it's prettier! Sorry about that. It's because I made a change to that doc directly in the browser and prettier never had a chance to run on it, so now it's being run on that file and showing changes.

**Comment by bhavyaab** [comment]
My master is up to date with origin and timeFormate is merged with the latest master, but it is not fixing the error.

**Comment by karenmrose** [comment]
If it's helpful, here is the code we re-wrote on the call:

```
const hours = parseInt(timeString.slice(0, 2))
	const minutes = parseInt(timeString.slice(2, 5))

	const hoursIn24HourFormat =
		timeString.slice(5, timeString.length) === 'pm'
			? hours + 12
			: (hours < 10 ? '0' : '') + hours

	const minutesIn2DigitFormat = minutes > 0 ? minutes : '00'

	return Number(`${hoursIn24HourFormat}${minutesIn2DigitFormat}`)
```
